src/initialization/InitElement.cpp

```diff
@@ -184,13 +184,38 @@ namespace detail
         {
             auto a = detail::query_alignment(pp_element);
 
-            amrex::ParticleReal psi, rc, g, K2;
+            amrex::ParticleReal psi, rc, g;
+            amrex::ParticleReal R = 1;
+            std::string location_str = "entry";  // default
+            std::string model_str = "linear";    // default
+
+            // The default values below are from eq (52) of K. Hwang and S. Y. Lee (2015)
+            amrex::ParticleReal pi = ablastr::constant::math::pi;
+            amrex::ParticleReal K0 = pi*pi / 6.0;
+            amrex::ParticleReal K1 = 0;
+            amrex::ParticleReal K2 = 1;
+            amrex::ParticleReal K3 = 1.0/6.0;
+            amrex::ParticleReal K4 = 0;
+            amrex::ParticleReal K5 = 0;
+            amrex::ParticleReal K6 = 0;
             pp_element.getWithParser("psi", psi);
             pp_element.getWithParser("rc", rc);
             pp_element.getWithParser("g", g);
-            pp_element.getWithParser("K2", K2);
-
-            m_lattice.emplace_back( DipEdge(psi, rc, g, K2, a["dx"], a["dy"], a["rotation_degree"], element_name) );
+            pp_element.queryAddWithParser("R", R);
+            pp_element.queryAddWithParser("K0", K0);
+            pp_element.queryAddWithParser("K1", K1);
+            pp_element.queryAddWithParser("K2", K2);
+            pp_element.queryAddWithParser("K3", K3);
+            pp_element.queryAddWithParser("K4", K4);
+            pp_element.queryAddWithParser("K5", K5);
+            pp_element.queryAddWithParser("K6", K6);
+
+            pp_element.queryAdd("model", model_str);
+            dipedge::Model const model = amrex::getEnum<dipedge::Model>(model_str);
+            pp_element.queryAdd("location", location_str);
+            dipedge::Location const location = amrex::getEnum<dipedge::Location>(location_str);
```

Comment:
@WeiqunZhang just checked in that we can read directly into an enum in ParmParse, too! :)

Reply:
Not exactly right now. `query` works, but not `queryAdd`. We can certainly support that.

Reply:
https://github.com/AMReX-Codes/amrex/pull/4765 adds `queryAdd` for amrex_enum.

Reply:
Awesome, thnak you! Let me refactor this after the PR, so we can do a 25.11 release depending on AMReX 25.11